File: src/longHourAndAHalf/Character.kt
package longHourAndAHalf

@Suppress("KDocUnresolvedReference")
/**
 * Main character that possessed by a player.
 *
 * **Usage:**
 *
 * Character is created in [startGame], then passed to the [Gameplay] constructor as a parameter.
 *
 * @constructor Creates a new character.
 *
 * [name], [gender] and [bladder] properties' values are copied from the
 * [profile] that passed as the constructor parameter.
 *
 * @author JavaBird
 *
 * @property underwear This character's underwear.
 * @property outerwear This character's outerwear.
 */
class Character(profile: CharacterProfile, val underwear: Underwear, val outerwear: Outerwear) {
    /**
     * Character's name.
     *
     * Name is used by NPCs when they talk to character.
     */
    val name = profile.name
    /**
     * Character's gender.
     *
     * Gender affects several game text fragments.
     */
    val gender = profile.gender

    /** Character's bladder. */
    val bladder = Bladder(0, profile.incontinence)
}

File: src/character.py
from bladder import Bladder


class CharacterProfile:
    def __init__(self, name, gender, incontinence):
        self.name = name
        self.gender = gender
        self.incontinence = incontinence


class Character:
    def __init__(self, profile, underwear, outerwear):
        self.name = profile.name
        self.gender = profile.gender
        self.bladder = Bladder(profile.incontinence)
        self.embarrassment = 0
        self.tummy_water = 0
        self.underwear = underwear
        self.outerwear = outerwear

    @property
    def dryness(self):
        return self.underwear.dryness + self.outerwear.dryness
